lib/spack/spack/util/module_cmd.py

```diff
@@ -162,7 +157,11 @@ def get_path_from_module(mod):
     # Read the module
     text = modulecmd('show', mod, output=str, error=str).split('\n')
 
-    return get_path_from_module_contents(text, mod)
+    p = get_path_from_module_contents(text, mod)
+    if not os.path.exists(p):
```

Comment:
@balay It turns out that `get_path_from_module_contents` can return `None` in certain cases (including the unit tests), which trips `os.path.exists`, so I think the test will need to be stricter:

`if p and (not os.path.exists(p)):`

(apologies for missing that)

Reply:
Thanks! pushed this fix.